### PR TITLE
feat: Theme Hacker Owl

### PR DESCRIPTION
**Pull request by koolamusic**
Hi Guys, created something here out of a little burning need, I'd be using remix a lot and thus wanted an IDE with a bit of a modern look and feel. 

I consider this continuous WIP, and would take responsibility for this theme even as I continue to use it for my work. 
Adds a monospace font and sans-serif and also, I'd find a way to map more `ace-identifier` keywords to the syntax highlight config. 

Also I create theme color variables using CSS Variables, so that the theme can easily be adapted. 
Feedback is welcome, like I said, this will be a continuous effort. 

Thanks

https://user-images.githubusercontent.com/8960757/131011195-957e65ba-0b99-46a3-b28d-fc6da2cc7afc.mov

### CODE REVIEW

File: apps/remix-ide/src/app/tabs/theme-module.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const _paq = window._paq = window._paq || []
 
 const themes = [
   { name: 'Dark', quality: 'dark', url: 'assets/css/themes/remix-dark_tvx1s2.css' },
+  { name: 'HackerOwl', quality: 'dark', url: 'assets/css/themes/remix-night_owl.css' },
```
Comment (LianaHus) [inline apps/remix-ide/src/app/tabs/theme-module.js:10]: Please put it after the Light theme

Comment (koolamusic) [reply apps/remix-ide/src/app/tabs/theme-module.js:10]: sure. will push an update

File: apps/remix-ide/src/app/tabs/theme-module.js [hunk 2]
```diff
@@ -8,10 +8,10 @@ const _paq = window._paq = window._paq || []
 const themes = [
   { name: 'Dark', quality: 'dark', url: 'assets/css/themes/remix-dark_tvx1s2.css' },
   { name: 'Light', quality: 'light', url: 'assets/css/themes/remix-light_powaqg.css' },
+  { name: 'HackerOwl', quality: 'dark', url: 'assets/css/themes/remix-hacker_owl.css' },
   { name: 'Midcentury', quality: 'light', url: 'assets/css/themes/remix-midcentury_hrzph3.css' },
   { name: 'Black', quality: 'dark', url: 'assets/css/themes/remix-black_undtds.css' },
   { name: 'Candy', quality: 'light', url: 'assets/css/themes/remix-candy_ikhg4m.css' },
-
```
Comment (LianaHus) [inline apps/remix-ide/src/app/tabs/theme-module.js:14]: the line is separating custom-made themes from ready ones but maybe it is fine to remove it

**Review by LianaHus — APPROVED** [review]

**Comment by LianaHus** [comment]
> @LianaHus had time to review?

looks good to me. thanks for your contribution! :)

**Review by LianaHus — CHANGES_REQUESTED** [review]
since there is no borders for sidepanel we need different colors for the background

**Comment by koolamusic** [comment]
> since there is no borders for sidepanel we need different colors for the background
> 
> Screenshot from 2021-09-06 11-19-20
> Screenshot from 2021-09-06 11-19-37
> Screenshot from 2021-09-06 11-20-04

Noted

**Comment by koolamusic** [comment]
Screenshot 2021-09-06 at 11-21-12 Remix - Ethereum IDE


@LianaHus added the patches.

**Comment by LianaHus** [comment]
> Screenshot 2021-09-06 at 11-21-12 Remix - Ethereum IDE
> 
> @LianaHus added the patches.

i still don't see the 'border' between switch tab and terminal

**Comment by koolamusic** [comment]
> > Screenshot 2021-09-06 at 11-21-12 Remix - Ethereum IDE
> > @LianaHus added the patches.
> 
> i still don't see the 'border' between switch tab and terminal

I feel like I am missing something. 
is there a way you could annotate the section.

**Comment by koolamusic** [comment]
image


@LianaHus see this

**Comment by LianaHus** [comment]
the contrast between right border of panel and bg is not enough to see it but it is visible anyway, so if you like it like that I don't mind

**Review by LianaHus — APPROVED** [review]

**Review by LianaHus — CHANGES_REQUESTED** [review]
I just realize you are using extra classes only for this them, can we avoid this?

**Comment by koolamusic** [comment]
ok, let me take a look at it

**Comment by koolamusic** [comment]
image


Added the fixes and changed the sidebar panel color to fix concerns with the contrast between the left and right panel.

**Comment by koolamusic** [comment]
@LianaHus can we please go over this again, what do I need to move this PR ?

**Comment by Hbutlercapone** [comment]
I'll merge all tonight I apoligize I had to move into a new home

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by koolamusic** [comment]
@Hbutlercapone do you need any assistance with this?

**Review by Hbutlercapone — APPROVED** [review]

**Review by Hbutlercapone — APPROVED** [review]
Approve merge of 24 commits for feat:Theme Hacker Owl #1527

**Comment by koolamusic** [comment]
> 

these issues have been resolved

**Comment by Hbutlercapone** [comment]
Its saying its pending two other reviewers reviews

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by koolamusic** [comment]
@ryestew <> @LianaHus  ^^

File: apps/remix-ide/src/app/editor/contextView.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    this._nodes = null
+    this._current = null
+    this.sourceMappingDecoder = sourceMappingDecoder
+    this.previousElement = null
+    this.contextualListener.event.register('contextChanged', nodes => {
+      this.show()
+      this._nodes = nodes
+      this.update()
+    })
+    this.contextualListener.event.register('stopHighlighting', () => {
```
Comment (ryestew) [inline apps/remix-ide/src/app/editor/contextView.js:37]: this line is triggering a linting error: Unexpected empty arrow function

**Review by ryestew — CHANGES_REQUESTED** [review]
Thanks so much for your update.  There is a linting error that was triggered which needs to be fixed.  We might have another comment in a day or two.  Almost there!

**Review by ryestew — CHANGES_REQUESTED** [review]
Please remove all changes to contextView.js and we will take care of that part. So the only files that should be in your pr are remix-hacker_owl.css and theme-module.js.  Then we can merge it!  Thanks

**Comment by Aniket-Engg** [comment]
> Please remove all changes to contextView.js and we will take care of that part. So the only files that should be in your pr are remix-hacker_owl.css and theme-module.js. Then we can merge it! Thanks

@koolamusic Can you consider this?

**Comment by koolamusic** [comment]
> > Please remove all changes to contextView.js and we will take care of that part. So the only files that should be in your pr are remix-hacker_owl.css and theme-module.js. Then we can merge it! Thanks
> 
> @koolamusic Can you consider this?

Yes. on it

**Review by ryestew — APPROVED** [review]
Thanks and thanks for your patience @koolamusic!!

**Comment by Aniket-Engg** [comment]
Thanks for your continuous co-operation and this great contribution @koolamusic !!!

**Comment by koolamusic** [comment]
My pleasure ❤️

**Pull request merged**